src/figuresClasses.ts

```diff
@@ -1,19 +1,84 @@
-export interface Figure {
-
+enum Shapes {
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+  Triangle = 'triangle',
 }
 
-export class Triangle {
+type Color = 'red' | 'green' | 'blue';
 
+function roundNumbers(num: number): number {
+  return Math.floor(num * 100) / 100;
 }
 
-export class Circle {
+export interface Figure {
+  shape: Shapes;
+  color: Color;
+  getArea(): number;
+}
 
+export class Circle implements Figure {
+  public shape = Shapes.Circle;
+
+  constructor(
+    public color: Color,
+    public a: number,
+  ) {
+    if (a <= 0) {
+      throw new Error('Incorrect radius');
+    }
+  }
+
+  getArea(): number {
+    return roundNumbers((this.a ** 2) * Math.PI);
+  }
 }
 
-export class Rectangle {
+export class Rectangle extends Circle {
+  public shape = Shapes.Rectangle;
+
+  constructor(
+    color: Color,
+    a: number,
+    public b: number,
+  ) {
+    super(color, a);
 
+    if (a <= 0 || b <= 0) {
+      throw new Error('This is not a rectangle');
+    }
+  }
+
+  getArea(): number {
+    return roundNumbers(this.a * this.b);
+  }
 }
 
-export function getInfo(figure) {
+export class Triangle extends Rectangle {
+  public shape = Shapes.Triangle;
+
+  constructor(
+    color: Color,
+    a: number,
+    b: number,
+    public c: number,
```

Comment:
better to keep figure-specific measures private (radius is private for circle, 3 sides - for triangle, etc.; while color is public - shared across all figures)
```suggestion
    private a: number,
    private b: number,
    private c: number,
```

Reply:
I think private doesn't give access to the extended classes so I had to use protected.